main.py

```diff
@@ -58,6 +59,9 @@ def run(filename, instance_as_str=True, rule_type=RuleType.ALL):
             print(f.read(), file=sys.stderr)
             exit(1)
         for item in log:
+            check_disabled = 'disabled' in item['tags']
```

Comment:
Good question. The necessity comes from the test_main.py file, more specifically this part (https://github.com/buildingSMART/ifc-gherkin-rules/blob/main/test/test_main.py#L31)

```
if base.startswith("fail-") and 'disabled' not in results:
    assert len(results) > 0
```

For every fail- file we check, if there are errors any yielded, expecting some errors. Having a tag filter in place no errors are yielded. 

So I decided to yield 'disabled', to cover it, while having a piece of clear information on what's happened.

Reply:
Ok, now I understand it a little bit better. 

I changed the solution to be more generic, now it returns a tuple in case of a disabled rule. Should be preferable